add onReady hook (#2166)
fastify/fastify: docs/Hooks.md

You can hook into the application-lifecycle as well.

- [onReady](#onready)
- [onClose](#onclose)
- [onRoute](#onroute)
- [onRegister](#onregister)

### onReady
Triggered before the server start listening for requests. It can not change the routes or add new hooks.

> The functions added to this hook will be waited. 

Reviewer: What do you mean? This is not clear

Author: Updated

The first argument is the `done` callback and the `this` binded to the function is the Fastify instance to which
the hook has been added.

```js
fastify.addHook('onReady', (done) => {
  // Some code
  this.validatorCompiler === fastify.validatorCompiler
  done()
})

> ```

Reviewer: Please add an example using async/await

You can hook into the application-lifecycle as well.

- [onReady](#onready)
- [onClose](#onclose)
- [onRoute](#onroute)
- [onRegister](#onregister)

### onReady
Triggered before the server starts listening for requests. It cannot change the routes or add new hooks.
Registered hook functions are executed serially. Only after all `onReady` hook functions have completed will the server start listening for requests.
Hook functions accept one argument: a callback, `done`, to be invoked after the hook function is complete. Hook functions are invoked with `this` bound to the associated Fastify instance.

```js
// callback style
fastify.addHook('onReady', (done) => {
  // Some code
  this.validatorCompiler === fastify.validatorCompiler

>   done()

Reviewer 2: Does this accept an `Error` to abort the server starting?

Author: Yes, it is

Fixed the readme 👌

You can hook into the application-lifecycle as well.

- [onReady](#onready)
- [onClose](#onclose)
- [onRoute](#onroute)
- [onRegister](#onregister)

### onReady
Triggered before the server starts listening for requests. It cannot change the routes or add new hooks.
Registered hook functions are executed serially. Only after all `onReady` hook functions have completed will the server start listening for requests.

> The first argument is the `done` callback and the `this` binded to the function is the Fastify instance to which

Author: ```suggestion
Hook functions accept one argument: a callback, `done`, to be invoked after the hook function is complete. Hook functions are invoked with `this` bound to the associated Fastify instance.
```

### onReady

> Triggered before the server start listening for requests. It can not change the routes or add new hooks.

Reviewer 2: ```suggestion
Triggered before the server starts listening for requests. It cannot change the routes or add new hooks.
```

> The hooks are executed serially and after the complete execution of them the server will listen.

Reviewer 2: ```suggestion
Registered hook functions are executed serially. Only after all `onReady` hook functions have completed will the server start listening for requests.
```

The first argument is the `done` callback and the `this` binded to the function is the Fastify instance to which

> the hook has been added.

Reviewer 2: ```suggestion
Hook functions accept one argument: a callback, `done`, to be invoked after the hook function is complete. Hook functions are invoked with `this` bound to the associated Fastify instance.
```
